### PR TITLE
[Ready] Added http.Client factory that creates client with default options

### PR DESCRIPTION
**Pull request by turboezh**
The problem: if we try to use `-yolo` option it will create new instance of http.Transport instead of `http.DefaultTransport` that used by default. But `http.DefaultTransport` use `ProxyFromEnvironment` to transparently handle proxy setting. As result we can't go through proxy anymore.

This factory will create a new transport with all default options from `http.DefaultTransport` except TLSClientConfig which is responsible for accepting invalid certificates. So the proxy handling will be kept.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -85,3 +82,18 @@ func open(name string, cli *http.Client) (io.ReadCloser, error) {
...
+	defaultTransport := http.DefaultTransport.(*http.Transport)
+	transport := &http.Transport{
+		Proxy:                 defaultTransport.Proxy,
+		DialContext:           defaultTransport.DialContext,
+		MaxIdleConns:          defaultTransport.MaxIdleConns,
+		IdleConnTimeout:       defaultTransport.IdleConnTimeout,
+		ExpectContinueTimeout: defaultTransport.ExpectContinueTimeout,
+		TLSHandshakeTimeout:   defaultTransport.TLSHandshakeTimeout,
+		TLSClientConfig:       &tls.Config{InsecureSkipVerify: insecure},
+	}
```
Comment (TheZeroSlave) [inline main.go]: m.b. use compiler copying?
defaultTransport := *(http.DefaultTransport.(*http.Transport))
defaultTransport. TLSClientConfig = &tls.Config{InsecureSkipVerify: insecure}
return &http.Client{Transport: transport}

Comment (turboezh) [reply main.go]: I've just copied that from StackOverflow. =) Your version is better, thanks.

**Comment by fiorix** [comment]
This is conflicting with the current master, probably because I stupidly merged y our other diff before merging this one.

I believe you have to rebase before we can continue here.

**Comment by fiorix** [comment]
👍

**Pull request merged**